monkey/monkey_island/cc/app.py

```diff
@@ -72,36 +73,54 @@ def serve_home():
     return serve_static_file(HOME_FILE)
 
 
-def init_app_config(app, mongo_url, data_dir: Path):
+def setup_authentication(app, data_dir):
+    flask_security_config = generate_flask_security_configuration(data_dir)
+
+    # TODO: After we switch to token base authentication investigate the purpose
+    # of `SECRET_KEY` and `SECURITY_PASSWORD_SALT`, take into consideration
+    # the discussion https://github.com/guardicore/monkey/pull/3006#discussion_r1116944571
+    app.config["SECRET_KEY"] = flask_security_config["secret_key"]
+    app.config["SECURITY_PASSWORD_SALT"] = flask_security_config["password_salt"]
+    app.config["SECURITY_USERNAME_ENABLE"] = True
+    app.config["SECURITY_USERNAME_REQUIRED"] = True
+    app.config["SECURITY_REGISTERABLE"] = True
+    app.config["WTF_CSRF_CHECK_DEFAULT"] = False
+    app.config["SECURITY_CSRF_IGNORE_UNAUTH_ENDPOINTS"] = True
```

Comment:
Investigate which one we need and add a comment why

Reply:
We need both of these to be defaults as this disables CSRF protection on forms and unauth endpoints.

Reply:
So auth endpoints will still require CSRF?

Reply:
https://flask-security-too.readthedocs.io/en/stable/patterns.html#csrf-enable-protection-for-session-auth-but-not-token-auth